src/net/cruhland/axioms/CoreAlgebra/Monoid.agda

```diff
@@ -0,0 +1,20 @@
+module net.cruhland.axioms.CoreAlgebra.Monoid where
+
+import net.cruhland.axioms.AbstractAlgebra as AA
+open import net.cruhland.axioms.Eq as Eq using (_≃_; Eq)
+  
+record Monoid (G : Set) : Set₁ where
+  infix 7 _⊙_
+  field
+    {{eq}} : Eq G  -- includes reflexive, symmetric, and transitive properties
+    _⊙_ : G → G → G
+    identity : G
+
+    -- properties of composition
+    {{⊙-substitutive}} : AA.Substitutive² {A = G} _⊙_ _≃_ _≃_  -- Eq subst with ⊙
+    {{⊙-associative}} : AA.Associative {A = G} _⊙_
+
+    -- proof of identity
+    {{is-identity}} : AA.Identity² {A = G} _⊙_ identity
+
+open Monoid public   -- this puts _⊙_ into scope
```

Comment:
You probably want to say `open Monoid {{...}} public` here. This will bring `_⊙_` and `identity` into scope with `Monoid` as an instance argument. Otherwise, you'd need to explicitly pass a `Monoid` record as an argument when calling those functions, or do an explicit `open Monoid M using (_⊙_; identity)`. You're doing something like that in `Group.agda`, I'll leave a comment showing an alternative.

Reply:
And actually, it's even better to limit exactly which fields you make public:

```agda
open Monoid {{...}} public using (_⊙_; identity)
```